app/extensions/email/__init__.py

```diff
@@ -105,63 +157,173 @@ def template(self, template, **kwargs):
+                pass
+
+        # Strip out unused leftover CSS and minify before sending
+        assert NEWLINE_TEMP_CODE not in transformed_html
+        transformed_html_ = transformed_html.replace('\n', NEWLINE_TEMP_CODE)
+        minified_css_html_ = re.sub(
+            r'<style type="text/css">.*</style>', '', transformed_html_
+        )
+        minified_css_html = minified_css_html_.replace(NEWLINE_TEMP_CODE, '\n')
+        minified_html = htmlmin.minify(
+            minified_css_html,
+            remove_comments=True,
+            remove_empty_space=True,
+            remove_all_empty_space=True,
+        )
+
+        # Add web fonts
+        webfonts = [
+            '<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Assistant:200|Chango|Molle:400i&display=swap">',
+            '<link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">',
+            '<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Source+Code+Pro&display=swap">',
+        ]
+        webfonts_html = ''.join(webfonts)
+        minified_html = minified_html.replace(
+            '</head>', '%s</head>' % (WEBFONTS_PLACEHOLDER_CODE,)
+        )
+        final_html = minified_html.replace(WEBFONTS_PLACEHOLDER_CODE, webfonts_html)
+        with open('email.latest.html', 'w') as temp:
+            temp.write(final_html)
+        self.html = final_html
+
     def attach(self, filepath, atatchment_name, attachment_type='image/png'):
         with current_app.open_resource(filepath) as asset:
             self.attach(atatchment_name, attachment_type, asset.read())
 
         return self
 
+    # note: in order to not get complex and have to break one Email up into multiple, we just use the first language
+    #   we find on a recipient; TODO develop a potential MultiLanguageEmail which is acually a (potential) list of Emails
+    def set_language(self):
+        if self.language or not self.recipients:
+            return
+
+        from app.modules.site_settings.models import SiteSetting
+        from app.modules.users.models import User
+
+        self.resolve_recipients()
+        for recip in self._original_recipients:
+            if isinstance(recip, User):
+                self.language = recip.get_preferred_langauge()
+                if self.language:
+                    return
+        self.language = SiteSetting.get_string('preferred_language', 'en_us')
+
+    def resolve_recipients(self):
+        if self._original_recipients:
+            return  # only do once
+        from app.modules.users.models import User
+
+        self._original_recipients = []
+        addresses = []
+        for recip in self.recipients:
+            self._original_recipients.append(recip)
+            if isinstance(recip, User):
+                addresses.append(recip.email)
+            else:
+                addresses.append(recip)
+        self.recipients = addresses
+
     def go(self, *args, **kwargs):
```

Comment:
go is a pretty horrible method name, 'send' would be better

Reply:
dont kill the messenger.  :grin: this was jp's original choice.   that said, i can _guess_ why he didnt use it -- there is a `send()` method in the Flask-Mail package already.  however, i _think_ it is on the Mail object, not the Message.  so maybe i can make a change here and actually use `send()`.... lemme confirm.